### PR TITLE
[MXNET-703] Use static libs for TensorRT

### PR DESCRIPTION
**Pull request by KellenSunderland**
## Description ##
Moving a few libraries to static linking to mitigate risk of dep conflicts.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

TODO: Fix setup.py

### CODE REVIEW

**Comment by lebeg** [comment]
Could you write how big the size difference is now? I mean the size of libmxnet.so

**Comment by KellenSunderland** [comment]
I've tried to have a good balance between portability and binary size.  What I'm really optimizing for in terms of file size is the wheel package size.  It's currently 77MB with these changes.  Before was 70 MB iirc.

**Comment by kalyc** [comment]
Thanks for your contribution @KellenSunderland 
Is the PR still WIP?

@mxnet-label-bot[pr-awaiting-review]

File: python/setup.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def config_cython():
         return []
 
 
-setup(name='mxnet',
+setup(name='mxnet-tensorrt-cu90',
```
Comment (lebeg) [inline python/setup.py]: This is WIP?

**Comment by KellenSunderland** [comment]
Yes still wip.

**Comment by vandanavk** [comment]
Continuing with pr-work-in-progress tag unless there is any change/update

cc @KellenSunderland

**Comment by KellenSunderland** [comment]
@vandanavk Should be ready for a review now, no long a wip.

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy Please change the label to pr-awaiting-review

**Comment by KellenSunderland** [comment]
@szha @nswamy Would one of you be able to have a look at this PR?

**Comment by Roshrini** [comment]
@KellenSunderland Can you please resolve conflicts?
@nswamy Can you please review this PR?

**Comment by ankkhedia** [comment]
@KellenSunderland It seems that there are still build failures. Could you please look into it?

**Comment by KellenSunderland** [comment]
@ankkhedia Looks like the build errors relate to the old findalgo function.  My best guess is that this PR (and others failing with this error) just need to be rebased.  I'll give it a shot when I have a few minutes.

**Comment by stu1130** [comment]
@lebeg @marcoabreu @larroy ping for review

**Comment by KellenSunderland** [comment]
Closing to prioritize other two TensorRT PRs which address bugs.

**Pull request closed without merging**